### PR TITLE
 [add] 新增deploy目录,组件deployment,更新README.md

### PR DESCRIPTION
**Pull request by jia9iZhang**
补充部署AutoML平台方式的初版

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,46 @@
...
+#### Step 3: 部署AutoML
+
+获取Deployment：
+
+- AutoML:[https://github.com/ml-for-good/automl/blob/main/Investigation/java/deploy/deployment.yaml](https://github.com/ml-for-good/automl/blob/main/Investigation/java/deploy/deployment.yaml)
+
+部署命令：
+
+```bash
+kubectl apply -f automl-deployment.yaml
```
Comment (free-y) [inline README.md:42]: I haven't  seen automl-deployment.yaml

Comment (jia9iZhang) [reply README.md:42]: automl-deployment.yaml should be the name of our final service deployment file, I reserve it now

Comment (free-y) [reply README.md:42]: the platform should be Installed with a single command

File: README.md [hunk 2]
```diff
@@ -1,5 +1,46 @@
...
+
+### How kubernetes deploys AutoML
+
+需要开放以下端口：Todo
+
+#### Step 1: 部署依赖环境
+
+- Minikube
+- Docker
+
```
Comment (free-y) [inline README.md:23]: By the way, could you please also add the steps about installing the prerequisites. Below is an example
https://github.com/ml-for-good/automl/blob/f0c6cd8fffbbdc962f043928cdc2a79596d98c99/Investigation/python/deploy/README.md

File: README.md (hunk 1 above)
Comment (free-y) [reply README.md:42]: Here is a demo for single command deployment https://github.com/kubeflow/manifests#install-with-a-single-command

Comment (free-y) [reply README.md:42]: for reference only https://nuclio.io/docs/latest/setup/minikube/getting-started-minikube/

Comment (free-y) [reply README.md:42]: one more reference https://docs.gitlab.com/charts/development/minikube/

**Comment by free-y** [comment]
Deploy to Kubernetes with or without helm
https://bothe.at/packaging-kubernetes-applications-with-helm/#How_you_deploy_to_Kubernetes_without_Helm
https://www.baeldung.com/ops/kubernetes-helm

File: README.md [hunk 3]
```diff
@@ -1,5 +1,131 @@
...
+#### Step 1: 部署依赖服务
+
+**MySQL部署**
+
+获取MySQL部署文件:
+
+- 获取mysql-pv.yaml：
+
+  ```bash
+  curl -O https://github.com/ml-for-good/automl/tree/main/deploy/mysql-pv.yaml
```
Comment (free-y) [inline README.md:47]: Currently we can focus on deploy the folder 'deploy', we don't need to download from remote. Just need to clone the repo, then cd into the repo folder, then run the deploy command.

**Comment by jia9iZhang** [comment]
yes

暖暖阳光 ***@***.***> 于2022年8月20日周六 11:32写道：

**Comment by free-y** [comment]
> Deploy to Kubernetes with or without helm https://bothe.at/packaging-kubernetes-applications-with-helm/#How_you_deploy_to_Kubernetes_without_Helm https://www.baeldung.com/ops/kubernetes-helm

And make the steps as just deploying the folder
image

File: README.md [hunk 4]
```diff
@@ -1,5 +1,64 @@
...
+
+### Start Deploy
+
+开始部署前,确保获取到最新部署文件!!!
+创建需要的**namespace**
+
+```bash
+#后续有需要再加
+kubectl create ns mysql
+kubectl create ns automl
```
Comment (free-y) [inline README.md:46]: I think automl and minio namespace could be removed currently.

File: README.md [hunk 5]
```diff
@@ -1,5 +1,61 @@
...
+  - 详细参考:https://minikube.sigs.k8s.io/docs/start/
+
+**Linux:** Todo
+
+**Windows:** Todo
+
+### Start Deploy
+
+开始部署前,确保获取到最新部署文件!!!
+创建需要的**namespace**
```
Comment (free-y) [inline README.md:41]: Please use yaml file to deploy namespace resource

File: deploy/mysql-pv.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  labels:
+    type: local
+spec:
+  storageClassName: manual
+  capacity:
+    storage: 20Gi
+  accessModes:
+    - ReadWriteOnce
+  hostPath:
+    path: "/mnt/data"
```
Comment (free-y) [inline deploy/mysql-pv.yaml:14]: How to make sure this path exists in the host machine?

File: deploy/mysql-pv.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: mysql-pv-volume
+  labels:
+    type: local
+spec:
+  storageClassName: manual
+  capacity:
+    storage: 20Gi
```
Comment (free-y) [inline deploy/mysql-pv.yaml:10]: Maybe we need to reduce the storage size since we will not allocate so much disk space to Minikube node.

Comment (jia9iZhang) [reply deploy/mysql-pv.yaml]: has been modified to 1G

**Pull request merged**